sympy/core/tests/test_match.py

```diff
@@ -709,6 +709,37 @@ def test_match_issue_17397():
     assert r == {a3: x - 4 + 4/x, b3: 1 - 2/x, c3: x - 4}
 
 
+def test_match_issue_21942_1():
```

Comment:
This test already passes in master. How has your change changed the behavior of match?

Reply:
Sorry, I got a bit carried away by trying to make the XFAIL look like this one. I'll update it with a test that actually makes a difference.